### ISSUE TITLE
[QUESTION] SKBitmap.Decode(stream) will dispose this stream, feature or bug?

### ISSUE THREAD

**Issue by xtuzy** [mono/SkiaSharp#2263]
I use SKBitmap.Decode(stream) to create a Bitmap, when i need continue use this stream, i found it is dispose.

![sharex-20220928152228](https://user-images.githubusercontent.com/17793881/192715483-3fd77411-1554-4c91-9593-4b7b80638eca.png)

![sharex-20220928152307](https://user-images.githubusercontent.com/17793881/192715523-64eb35f2-ddab-4857-bb85-05f2b435c731.png)

Skiasharp Version: 2.88.2

**Comment by LukePulverenti** [comment]
Are you saying this was different in earlier versions?

**Comment by xtuzy** [comment]
no difference compare with 2.88.0, i  just feel it is not right, stream should be disposed by user like this
```
using(var stream = ***)
{
  var bitmap = SKBitmap.Decode(stream);
}
```

**Comment by dallonxu** [comment]
@xtuzy 

Just use `SKData` to wrap the `stream`, the `stream` will not be closed after `SKBitmap.Decode`.
```cs
using(var stream = ***)
{
    var skData = SKData.Create(stream);
    var bitmap = SKBitmap.Decode(skData);
    //stream will not be closed here
}
```

**Comment by mattleibow** [comment]
I'll need to confirm, but I am pretty sure this is expected. The decode usually reads all the way to the end, and the stream may not always be rewindable. So there was not much point in keeping it open... However, the Stream overloads could maybe get an additional parameter to indicate whether or not to close the stream.

**Comment by jonmdev** [comment]
I agree this makes no sense. What if we want to use the stream for something else after this function? We have lost it and must get it again. It is wasteful.

**Comment by nifiz** [comment]
@mattleibow Managing a Stream should be left up to the user, not a completely different function they call.

**Comment by tigransard** [comment]
there is another workaround that I fond in the code

var skStream = new SKManagedStream(fileStream, disposeManagedStream: false);
using var original = SKBitmap.Decode(skStream);

in this case your original Stream will not be closed;

**Comment by mattleibow** [comment]
I think we could ad some internal wrapper/proxy stream so you can disable this feature.

Skia will always dispose, but we can hide it if we want.

@tigransard has a good workaround that we could implement internally.
